projects/vdk-core/tests/functional/run/test_run_templates.py

```diff
@@ -59,3 +58,38 @@ def test_run_job_with_template_error(tmpdir: py.path.local):
 
     cli_assert_equal(1, result)
     assert "no such table: no_such_table" in result.output
+
+
+def test_run_job_with_cancelled_template():
+    """
+    Test runs a data job with two steps which in turns runs a template with two steps.
+    Every step prints e.g Step 1,2 and Template Step 1,2 etc. After the first template
+    step is executed the cancel_job_execution() method is called. Test aims to check
+    correct template cancellation (when a template step is cancelled remaining data job
+    steps continue)
+
+    Test checks:
+    template execution is canceled and all job steps are executed.
+    template steps prior to the cancel_job_execution() call are executed.
+    proper exit code is set.
+    proper message is logged confirming template was cancelled from cancel_job_execution() method.
+    :return:
+    """
+    test_runner = CliEntryBasedTestRunner(
+        TemplatePlugin(
+            "template-cancel", pathlib.Path(util.job_path("template-cancel"))
```

Comment:
nit: though. this is also fine , you can also reuse "job-cancel" as a template here instead of creating new job "template-cancel"